docs/snapshots.rst

```diff
@@ -0,0 +1,75 @@
+.. -*- coding: utf-8 -*-
+
+.. _snapshots:
+
+
+Snapshots
+=========
+
+Information about files in a Magic Folder are encapsulated in a
+logical "Snapshot" object. We distinguish between a "local" Snapshot
+and a "remote" Snapshot (in code, these are `LocalSnapshot` and
+`RemoteSnapshot`).
+
+The difference is that a Remote Snapshot has been signed and uploaded
+into the Tahoe-LAFS Grid. A Local Snapshot exists only on one device's
+local disk (so other participants cannot see it yet).
+
+A Snapshot has one or more parents, forming a Directed Asyclic Graph
+(DAG). Snapshots are signed "as" they are uploaded into the Grid:
+first the actual content is uploaded as an immutable, yielding a
+read-only capability-string; then this string and other metadata is
+signed. The signature, metadata and pointer to the content constitutes
+the Remote Snapshot and itself is uploaded into the Tahoe Grid,
+represented as an immutable directory.
```

Comment:
This section might benefit from explaining some more of the "why" to go along with the "what".